### PR TITLE
PP-11314 pipeline for versioning node packages

### PR DESCRIPTION
**Pull request by nlsteers**
### WHAT

Limitations in the GitHub permissions model mean that if we want to automatically version and release our Node packages we'll need to do the versioning part from Concourse.

This pipeline monitors commits to the main branches of our `pay-js` packages. When a new commit (that is not a release commit created by this pipeline) is detected. Concourse will build and version the package appropriately, creating a release tag and opening a PR to merge in files affected by the `npm version patch` command.

### CODE REVIEW

File: ci/pipelines/pay-js.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          run:
+            path: sh
+            dir: js-metrics-git-release
+            args:
+              - -c
+              - |
+                apk add --no-cache git github-cli
+                
+                git remote remove origin
+                git remote add origin https://$GITHUB_TOKEN@github.com/$REPO
```
Comment (katstevens) [inline ci/pipelines/pay-js.yml:72]: Can we make sure the `$GITHUB_TOKEN` isn't logged out anywhere visible

Comment (nlsteers) [reply ci/pipelines/pay-js.yml:72]: It's not visible in the build output.

**Comment by katstevens** [comment]
Happy with how this is looking now 👍 When you're ready to put the temporary branch back to `master` I'll approve

File: ci/tasks/npm-version-and-create-pr.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+platform: linux
+image_resource:
+  type: docker-image
+  source:
+    repository: node
+    tag: alpine3.16
```
Comment (jfharden) [inline ci/tasks/npm-version-and-create-pr.yml:6]: Can you make this tag `current-alpine`

**Review by jfharden — APPROVED** [review]
❤️ Amazing work man!

**Pull request merged**